src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,899 @@
+      const stack: Array<number> = [];
+      appendRecursive(id, stack);
+      stack.reverse();
+      return stack;
+    });
+  }
+
+  for (const sample of samples) {
+    sample.backtraceStack = backtraceIDtoStack.get(sample.backtraceID);
+  }
+
+  return Promise.resolve(samples);
+}
+
+// This function reads the 'form.template' file which contains all the important information about
+// addressToFrameMap and Instruments' version
+async function readFormTemplateFile(tree) {
+  const formTemplate = tree.files.get('form.template');
+  const archive =
+    typeof formTemplate !== 'undefined'
+      ? readInstrumentsArchive(await readAsArrayBuffer(formTemplate))
+      : {};
+
+  const version = archive['com.apple.xray.owner.template.version'];
+  let selectedRunNumber = 1;
+  if ('com.apple.xray.owner.template' in archive) {
+    selectedRunNumber = archive['com.apple.xray.owner.template'].get(
+      '_selectedRunNumber'
+    );
+  }
+  let instrument = archive.$1;
+  if ('stubInfoByUUID' in archive) {
+    instrument = Array.from(archive.stubInfoByUUID.keys())[0];
+  }
+  const allRunData = archive['com.apple.xray.run.data'];
+
+  const runs: FormTemplateRunData[] = [];
+  for (const runNumber of allRunData.runNumbers) {
+    const runData = getOrThrow(allRunData.runData, runNumber);
+    const symbolsByPid = getOrThrow(runData, 'symbolsByPid');
+
+    const addressToFrameMap = new Map<number, FrameInfo>();
+    for (const symbols of symbolsByPid.values()) {
+      for (const symbol of symbols.symbols) {
+        if (!symbol) continue;
+        const { sourcePath, symbolName, addressToLine } = symbol;
+        for (const address of addressToLine.keys()) {
+          getOrInsert(addressToFrameMap, address, () => {
+            const name = symbolName || `0x${zeroPad(address.toString(16), 16)}`;
+            const frame: FrameInfo = {
+              key: `${sourcePath}:${name}`,
+              name: name,
+            };
+            if (sourcePath) {
+              frame.file = sourcePath;
+            }
+            return frame;
+          });
+        }
+      }
+
+      runs.push({
+        number: runNumber,
+        addressToFrameMap,
+      });
+    }
+  }
+
+  return {
+    version,
+    instrument,
+    selectedRunNumber,
+    runs,
+  };
+}
+
+// This function returns a directory tree where each node of tree
+// is a object consist of name, files and subdirectories fields
+async function extractDirectoryTree(entry: {
+  name: string,
+}): Promise<TraceDirectoryTree> {
+  const node = {
+    name: entry.name,
+    files: new Map(),
+    subdirectories: new Map(),
+  };
+
+  const children = await new Promise((resolve, reject) => {
+    // FileSystemDirectoryEntry.createReader() is a non-standard function for now, I have checked in Firefox and Chrome. It's working fine in it.
+    // We will remove FlowFixMe once it becomes a standard function
+    // $FlowFixMe createReader is not present in entry
```

Comment:
Another issue(or the same one with the other `FlowFixMe`) for that fixme and referencing that here would be great.